lib/Net/Amazon/S3/Client/Object.pm

```diff
@@ -355,6 +355,74 @@ sub query_string_authentication_uri {
     )->query_string_authentication_uri( $self->expires->epoch, $query_form );
 }
 
+sub head {
+    my $self = shift;
+
+    my $http_request =
+        Net::Amazon::S3::Request::GetObject->new(
+            s3     => $self->client->s3,
+            bucket => $self->bucket->name,
+            key    => $self->key,
+            method => 'HEAD',
+        )->http_request;
+
+    my $http_response = $self->client->_send_request($http_request);
+
+    confess 'Error head-object ' . $http_response->as_string
+        if $http_response->code != 200;
+
+    my %metadata;
+    my $headers = $http_response->headers;
+    foreach my $name ($headers->header_field_names) {
+        my $lc_name = lc($name);
+        if (
+            $lc_name =~ s/^x-amz-//
+                or $lc_name =~ m/^content-/
+                or $lc_name eq 'accept-ranges'
+                or $lc_name eq 'cache-control'
+                or $lc_name eq 'etag'
+                or $lc_name eq 'expires'
+                or $lc_name eq 'last-modified') {
+            next if ($lc_name eq 'id-2');
```

Comment:
Code readability note: this line looks like noop unless you notice slightly different L379.
Please do not mix substitution with match in one expression.

What about moving condition into extra method, eg: `_is_metadata_header` and L388 + L379 into `_format_metadata_name` ?

```
my %valid_metadata_headers = map +($_ => 1), (
  'accept-ranges',
  'cache-control',
  'etag',
  'expires',
  'last-modified',
);

sub _is_metadata_header {
  my ($self, $header) = @_;
  $header = lc $header;

  return 1 if exists $valid_metadata_headers{$header};
  return 1 if $header =~ m/^ x-amx- (?! id-2$) /x;
  return 1 if $header =~ m/^content-/;
  return 0;
}
```

Reply:
Ok. But should be the scope of %valid_metadata_headers the entire file? Otherwise, I can move the variable within _is_metadata_header.

Thank you for your suggests,
Emanuele